generics/helpers/elasticSearch.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * name : elasticSearch.js
+ * author : Aman Jung Karki
+ * created-date : 11-Jun-2020
+ * Description : elastic search common functionality
+ */
+
+/**
+  * Get user data
+  * @function
+  * @name get
+  * @param {Object} data
+  * @param {String} data.id - user id
+  * @param {String} data.index - elastic search index
+  * @param {String} data.type - elastic search type
+  * @returns {Object} Get user data.
+*/
+
+var get = function (data) {
+
+  return new Promise(async function (resolve, reject) {
+    try {
+
+      if (!data.id) {
+        throw "id is required";
+      }
+
+      if (!data.index) {
+        throw "index is required";
+      }
+
+      if (!data.type) {
+        throw "type is required";
+      }
+
+      const result = await elasticsearch.client.get({
+        id: data.id,
+        index: data.index,
+        type: data.type
+      }, {
+          ignore: [httpStatusCode["not_found"].status],
+          maxRetries: 3
+        });
+
+      return resolve(result);
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+};
+
+/**
+  * Create or update based on user data.
+  * @function
+  * @name createOrUpdate
+  * @param {Object} data - Data to be created or updated.
+  * @param {String} data.id - user id.
+  * @param {String} data.index - elastic search index.
+  * @param {String} data.type - elastic search type.
+  * @param {Object} data.body - updated or created body data.
+  * @param {String} [update = false] - if update is true update existing user data else create new user data.
+  * @returns {Object} created or updated user data.
+*/
+
+var createOrUpdate = function (data, update = false) {
+
+  return new Promise(async function (resolve, reject) {
+    try {
+
+      if (!data.id) {
+        throw "id is required";
+      }
+
+      if (!data.index) {
+        throw "index is required";
+      }
+
+      if (!data.type) {
+        throw "type is required";
+      }
+
+      if (!data.body) {
+        throw "body is required";
+      }
+
+      let result;
+
+      if (update) {
+        result = await elasticsearch.client.update({
```

Comment:
@aman-tunerlabs - Refer to Kendra ES function of create or update

`
/**
  * Create Or Update Operation.
  * @function
  * @name createOrUpdateDocumentInIndex
  * @param {String} id - (Non - Mandatory) Document ID.
  * @param {String} index - Index from which document is to deleted
  * @param {String} type - Type from which document is to deleted
  * @param {Object} data - Document Data to be created.
  * @returns {Promise} returns a promise.
*/

var createOrUpdateDocumentInIndex = function (index = "", type = "", id = "", data = "") {

  return new Promise(async function (resolve, reject) {
    try {

      if (index == "") {
        throw new Error("Index is required");
      }

      if (type == "") {
        throw new Error("Type is required");
      }

      if (id == "") {
        throw new Error("ID is required");
      }

      if (Object.keys(data).length == 0) {
        throw new Error("Data is required");
      }
      
      let documentObject = {
        id: id,
        index: index,
        type: type,
        body: {
          doc : data,
          doc_as_upsert : true,
        },
        refresh : true
      }

      let result = await elasticsearch.client.update(documentObject);

      return resolve(result);

    } catch (error) {
      return reject(error);
    }
  })
};`